core/controllers/base.py

```diff
@@ -160,11 +157,10 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
             self.values['user_email'] = user_settings.email
             self.user_id = user_settings.user_id
 
-            self.user_is_scheduled_for_deletion = user_settings.deleted
-
-            if (self.REDIRECT_UNFINISHED_SIGNUPS and not
+            if user_settings.deleted:
+                self.user_is_scheduled_for_deletion = user_settings.deleted
```

Comment:
Why is this changed from what it was before? I'm not sure why it needs to go into an "if" statement -- we should still be initializing it to False otherwise, right?

Reply:
We need this `if` so that the `if` on the next line can be `elif`, because the next `if` contains `user_services.has_fully_registered(user_settings.user_id)` which will fail if the user is scheduled for deletion.

Reply:
OK -- should we then set self.user_is_scheduled_for_deletion to False at the outset, or in the other branches?

Reply:
I think it is okay to set it at the outset, a similar thing is done with `self.partially_logged_in`.